hs_file_types/views.py

```diff
@@ -524,10 +536,21 @@ def add_metadata_element(request, hs_file_type, file_type_id, element_name, **kw
                               'element_id': element.id,
                               'metadata_status': metadata_status}
 
-        if logical_file.type_name() in ("NetCDFLogicalFile", "TimeSeriesLogicalFile"):
+        update_file = False
+        if logical_file.type_name() == "NetCDFLogicalFile":
+            # only spatial coverage creation does not need update of the nc file
+            if element_name.lower() != 'coverage':
+                update_file = True
+            elif element.type == 'period':
+                update_file = True
+        elif logical_file.type_name() == "TimeSeriesLogicalFile":
+            update_file = True
+
+        if update_file:
             logical_file.metadata.is_update_file = True
             logical_file.metadata.save()
-            ajax_response_data['is_update_file'] = logical_file.metadata.is_update_file
+
+        ajax_response_data['is_update_file'] = update_file
```

Comment:
This code block to set update_file is the same in add_metadata_element and update_metadata_element. It is better to remove duplication by putting this in a common private function to be called in both places.

Reply:
Good catch. Made the change.